streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java

```diff
@@ -554,12 +554,12 @@ public void shouldInvokeUserDefinedGlobalStateRestoreListener() throws Exception
         final Map<String, Object> kafkaStreams1Configuration = mkMap(
             mkEntry(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory(appId).getPath() + "-ks1"),
             mkEntry(StreamsConfig.CLIENT_ID_CONFIG, appId + "-ks1"),
-            mkEntry(StreamsConfig.restoreConsumerPrefix(ConsumerConfig.MAX_POLL_RECORDS_CONFIG), 1)
+            mkEntry(StreamsConfig.restoreConsumerPrefix(ConsumerConfig.MAX_POLL_RECORDS_CONFIG), 5)
```

Comment:
Given the latest changes, do we still need to increase this to `5`? Or should we keep it at `1`?

Reply:
```
[2024-09-06 20:37:07,333] INFO stream-thread [shouldInvokeUserDefinedGlobalStateRestoreListenerb1eh_2hFQ2CBtdaI_RxhgA-ks1-StreamThread-1] Finished restoring changelog shouldInvokeUserDefinedGlobalStateRestoreListenerb1eh_2hFQ2CBtdaI_RxhgA-KSTREAM-REDUCE-STATE-STORE-0000000001-changelog-0 to store KSTREAM-REDUCE-STATE-STORE-0000000001 with a total number of 18 records (org.apache.kafka.streams.processor.internals.StoreChangelogReader:693)
[2024-09-06 20:37:07,336] INFO stream-thread [shouldInvokeUserDefinedGlobalStateRestoreListenerb1eh_2hFQ2CBtdaI_RxhgA-ks1-StreamThread-1] Finished restoring changelog shouldInvokeUserDefinedGlobalStateRestoreListenerb1eh_2hFQ2CBtdaI_RxhgA-KSTREAM-REDUCE-STATE-STORE-0000000001-changelog-2 to store KSTREAM-REDUCE-STATE-STORE-0000000001 with a total number of 16 records (org.apache.kafka.streams.processor.internals.StoreChangelogReader:693)
[2024-09-06 20:37:07,339] INFO stream-thread [shouldInvokeUserDefinedGlobalStateRestoreListenerb1eh_2hFQ2CBtdaI_RxhgA-ks1-StreamThread-1] Finished restoring changelog shouldInvokeUserDefinedGlobalStateRestoreListenerb1eh_2hFQ2CBtdaI_RxhgA-KSTREAM-REDUCE-STATE-STORE-0000000001-changelog-4 to store KSTREAM-REDUCE-STATE-STORE-0000000001 with a total number of 19 records (org.apache.kafka.streams.processor.internals.StoreChangelogReader:693)
[2024-09-06 20:37:07,361] INFO stream-thread [shouldInvokeUserDefinedGlobalStateRestoreListenerb1eh_2hFQ2CBtdaI_RxhgA-ks2-StreamThread-1] Finished restoring changelog shouldInvokeUserDefinedGlobalStateRestoreListenerb1eh_2hFQ2CBtdaI_RxhgA-KSTREAM-REDUCE-STATE-STORE-0000000001-changelog-3 to store KSTREAM-REDUCE-STATE-STORE-0000000001 with a total number of 30 records (org.apache.kafka.streams.processor.internals.StoreChangelogReader:693)
[2024-09-06 20:37:07,865] INFO stream-thread [shouldInvokeUserDefinedGlobalStateRestoreListenerb1eh_2hFQ2CBtdaI_RxhgA-ks2-StreamThread-1] Finished restoring changelog shouldInvokeUserDefinedGlobalStateRestoreListenerb1eh_2hFQ2CBtdaI_RxhgA-KSTREAM-REDUCE-STATE-STORE-0000000001-changelog-1 to store KSTREAM-REDUCE-STATE-STORE-0000000001 with a total number of 17 records (org.apache.kafka.streams.processor.internals.StoreChangelogReader:693)
```

+1 to keep it at `1` as the number of records to restore gets reduced due to latest changes. Having enough loops/times to wait partitions migration (to stream_2) is better.